test/integration/tracking/TrackerWPDE.spec.ts

```diff
@@ -51,47 +65,55 @@ describe( 'TrackerWPDE', function () {
 	} );
 
 	it( 'collects tracking events until the tracker becomes available', async () => {
-		vi.useFakeTimers();
+		const tempTracker = window.TestTracker;
+		const trackingQueue: Function[] = [];
+		window.TestTracker = undefined;
 
 		const tracker = new TrackerWPDE(
 			'TestTracker',
 			'TestBanner05',
 			new Map<string, number>( [ [ 'action-1', 1 ], [ 'action-2', 1 ] ] )
 		);
 
+		Object.defineProperty( tracker, '_tracker', { writable: true, configurable: true, value: tempTracker } );
```

Comment:
This test has a bit more knowledge of the actual implementation than I'm comfortable with, testing implementation details instead of just testing behavior. While reviewing, I was also confused what the tests actually does. I'd suggest the following changes:
- rename `tempTracker` to `oldTracker` or `savedTestTracker` or something else, to make it clear in the "arrange" part that we're storing the default one for later.
- override `_tracker` with another tracker, maybe called "trackerSpy". Making sure that it isn't called supports the thing that's being tested, so I'd keep the `notHaveBeenCalled` check
- Drop checking the queue. We don't need to know how the implementation queues the calls.

Reply:
Yeah, I think the tests expose that the actual implementation isn't great. We should be injecting in a `PageTrackerFinder` or something so we have total control when the "real" page tracker appears. It'll have BBCs so I was reluctant to do it. We should put it on our list for next year.

`tempTracker` is for checking that the internal `_tracker` wasn't called befor the "real" one was found so maybe a better name would be `nullTrackerSpy` or `missingTrackerSpy`?

Reply:
Take a look at those changes.